compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -274,6 +274,7 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
         error: &SelectionError<'tcx>,
         fallback_has_occurred: bool,
     ) {
+        self.set_tainted_by_errors();
```

Comment:
Does this change anything?

Reply:
yes. It avoids ICEs and bogus errors in CTFE.